### DISCUSSION TITLE
Unexpected token < in JSON at position 0

### DISCUSSION THREAD

**Discussion by Rodagave** [requarks/wiki#5439] [Error / Bug Report] [answered]
Greetings. I have been facing this issue multiple times already whilst attempting to modify a page. I am not sure what could be causing this issue, as there's always something different that causes it, and sometimes deleting some lines fixes it, though sometimes the issue persists. 
![Screenshot 2022-07-02 173652](https://user-images.githubusercontent.com/49004312/177017978-831d8ec6-efdb-4bd9-9913-532678d6244a.png)
![Screenshot 2022-07-02 174028](https://user-images.githubusercontent.com/49004312/177018008-841a8f5c-78d9-44ac-8821-d8f564224fe6.png)

I'd highly appreciate it if this could be looked into.

**Comment by jarrodlovas** [comment]
Have you done any CSS/HTML overwriting? and are you trying to use the API for anything?

**Comment by Rodagave** [comment]
I have not performed any CSS nor HTML overwriting. Also, we are using wiki js' API.

**Reply by jarrodlovas** [reply]
Did you modify the app.js file located in the `client\themes\default\js` folder.
Because the error is saying there is an invalid  `<` at position 0 of your app.js file.

**Comment by Rodagave** [comment]
I haven't modified that file either. I've done some research and apparently, I am not the only one who has faced this issue.

**Comment by chapb** [comment]
I'm experiencing the same issue as well, running in Docker behind Traefik.

**Reply by chapb** [reply]
Actually, my issue is slightly different `wikijs Network error: JSON.parse: unexpected character at line 1 column 1 of the JSON data`.

Looks like:

- #5325 
- #5410

**Reply by chapb** [reply]
Also #3244

**Reply by chapb** [reply]
Sorry to hijack your issue. 

I was experiencing hanging at random intervals after container start with Wiki.js in Docker with Traefik as a reverse-proxy. 

If I was browsing the wiki, I'd get a 504 timeout. If I was editing the wiki, I'd get the JSON.parse error. The wiki would once again respond several minutes later without intervention, or when the container was restarted. I think these issues have the same root cause (a disconnect between the container and the reverse-proxy Traefik), just presenting differently depending on what I was doing at the time. 

Two changes seem to have reduced the occurrence of the error so far:

- Setting JWT audience to my wiki's URL (I don't think this changed anything)
- Specifying the Docker network on which Wiki.js contacts Traefik.

My uneducated guess would be that Traefik refreshed its connection to the container at an interval, and the Wiki.js container was swapping to the internal network (connecting to the postgres database) during that refresh.

**Comment by NGPixel** [comment, marked as answer]
This error is a generic network failure error, which is usually caused by a reverse-proxy (such as Cloudflare, nginx, Traefic, etc.) blocking the request. For example, trying to save a page with content like scripts or other special characters can trigger some reverse-proxy firewalls.

Try white-listing your IP and make sure your reverse-proxy config is correct so that requests don't get blocked and routed incorrectly.

**Reply by chapb** [reply]
Thanks!

**Reply by chapb** [reply]
BTW, I'd happily contribute a FAQ about this, or a working example of a configuration with Traefik, for the documentation, if so desired.

**Reply by NGPixel** [reply]
That'd be great. Feel free to write one at https://github.com/requarks/wiki/discussions/categories/guides
and I'll add a link to it in the docs.
